lib/AST/ProtocolConformanceRef.cpp

```diff
@@ -293,25 +301,40 @@ ProtocolConformanceRef::getAssociatedConformance(Type conformingType,
                      conformingType->is<UnresolvedType>() ||
                      conformingType->is<PlaceholderType>());
 
-  return ProtocolConformanceRef(protocol);
+  return ProtocolConformanceRef::forAbstract(conformingType, protocol);
```

Comment:
Oh I forgot about this spot!

Reply:
The above conformingType assert should eventually be in forAbstract() itself -- but maybe not yet

Reply:
Yeah, I want to get the refactoring in place first, then crank up the assertions until we can start simplifying.